### PR TITLE
Move all realm.io links to mongodb docs

### LINKED ISSUE #2143: Move all realm.io links to mongodb docs
Grepping for https://realm.io indicates we have a bunch of places where we're linking to the old docs. While those links are still valid, we should replace them with the corresponding links from the mongodb docs.

### PR DESCRIPTION
**Pull request by LaPeste**
## Description
Move all realm.io links to mongodb docs

I couldn't find some information, I was thinking to contact the docs team to ask. If you @nirinchev know already that some should not exist anymore, then I'll just rearrange the documents to rephrase/remove the links.

The changes were made on entries coming from the following command:
`grep -rin --exclude-dir={Docs,obj,bin,realm-core} "https://realm.io*" .`
executed at the highest level of realm-dotnet folder.

Fixes #2143 

##  TODO
* [ ] Changelog entry  --> I don't believe there's anything to write

### CODE REVIEW

File: Realm/Realm/Schema/RealmSchema.cs [hunk 1]
```diff
@@ -213,7 +213,7 @@ public RealmSchema Build()
                 if (Count == 0)
                 {
                     throw new InvalidOperationException(
-                        "No RealmObjects. Has linker stripped them? See https://realm.io/docs/xamarin/latest/#linker-stripped-schema");
+                        "No RealmObjects. Has linker stripped them? See https://thereIsntSuchDocYet");
```
Comment (nirinchev) [inline Realm/Realm/Schema/RealmSchema.cs:216]: Feel free to link to the legacy docs and open a ticket for the team to move the information to the new docs.

File: SUPPORT.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ The Realm team is here to help you with your Realm-related issues!
 
 ## Documentation
 
-Before asking questions, please familiarize yourself with our [Xamarin (.NET)](https://realm.io/docs/xamarin/latest/) documentation. We also have a number of [Tech Notes](https://realm.io/docs/tech-notes/) which cover various topics that may be of interest.
+Before asking questions, please familiarize yourself with our [.NET SDK](https://docs.mongodb.com/realm/dotnet/) documentation. We also have a number of [Tech Notes](https://IDontThinkThisExists) which cover various topics that may be of interest.
```
Comment (nirinchev) [inline SUPPORT.md:7]: Remove the tech notes sentence - we don't have those currently.

**Review by nirinchev — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -2,23 +2,23 @@
...
 
 ## Features
 
 * **Mobile-first:** Realm is the first database built from the ground up to run directly inside phones, tablets, and wearables.
-* **Simple:** Data is directly [exposed as objects](https://realm.io/docs/dotnet/latest/#models) and [queryable by code](https://realm.io/docs/dotnet/latest/#queries), removing the need for ORM's riddled with performance & maintenance issues. Plus, we've worked hard to [keep our API down to just 2 common classes](https://realm.io/docs/dotnet/latest/api/) (RealmObject and Realm): most of our users pick it up intuitively, getting simple apps up & running in minutes.
+* **Simple:** Data is directly [exposed as objects](https://docs.mongodb.com/realm/dotnet/objects/) and [queryable by code](https://docs.mongodb.com/realm/dotnet/query-engine/), removing the need for ORM's riddled with performance & maintenance issues. Plus, we've worked hard to [keep our API down to just 2 common classes](https://docs.mongodb.com/realm-sdks/dotnet/latest/) (RealmObject and Realm): most of our users pick it up intuitively, getting simple apps up & running in minutes.
```
[the remark this answers is not in the document]
Comment (LaPeste) [reply README.md:10]: Maybe that should be reflected also in "The main classes you will use are" section (4th line) in the documentation? ==> https://docs.mongodb.com/realm-sdks/dotnet/latest/

File: Realm/Realm.Unity/package.json [hunk 4]
```diff
@@ -14,8 +14,7 @@
       "mongodb"
     ],
     "author": {
-      "name": "Realm",
-      "email": "help@realm.io",
```
[the remark this answers is not in the document]
Comment (LaPeste) [reply Realm/Realm.Unity/package.json:18]: I'm waiting for the email. But only Brian knows and he's in holiday, but we expect to have one soon (2/3 days max).

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (nirinchev) [reply README.md]: We should completely replace/rewrite that one, just haven't had the time to look into it.

File: Realm/Realm.Unity/package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (nirinchev) [reply Realm/Realm.Unity/package.json:18]: You can also ping Sergey. As far as I remember, help@realm.io also works, just wondering if there's a better one to use.

[the remark this answers is not in the document]
Comment (bmunkholm) [reply Realm/Realm.Unity/package.json:18]: realm-help@mongodb.com

[the remark this answers is not in the document]
Comment (nirinchev) [reply Realm/Realm.Unity/package.json:18]: Brian, go build a snowman or something 😛 I explicitly didn't tag you so as not to disturb your vacationing!

Appreciate the help though 🙇‍♂️

**Review by nirinchev — APPROVED** [review]

**Pull request merged**